### PR TITLE
.github/workflows/setupPythonVenv.yml: added workflow

### PR DESCRIPTION
**Pull request by tatarurzvn**

### CODE REVIEW

**Comment by eduardostalinho** [comment]
Can you provide an evidence of this workflow running in another respository for an example on this PR?

Can you also add a README for it?

File: README.md [hunk 1]
```diff
@@ -177,3 +177,18 @@ jobs:
...
       SECRET_BUILD_ARGS: "TEAM_GITHUB_TOKEN=${ secrets.AUTOMATE_GITHUB_ACCESS_TOKEN }}"
 ```
+
+## Setup python env (setupPythonVenv.yml)
+
+In order to reduce clutter across all the projects on CC which use python runtime, we added this reusable action which caches the whole venv based on the `requirments.txt` file. 
+
+Usage in repos:
+```
+  setup_venv:
```
Comment (eduardostalinho) [inline README.md:187]: Can you provide an example of this running on service repository?

File: .github/workflows/setupPythonVenv.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      with:
+        python-version: ${{ inputs.python_version }}
+
+    - name: Setup Pip SSH Key
+      uses: webfactory/ssh-agent@v0.5.3
+      with:
+        ssh-private-key: ${{ secrets.pip_ssh_key }}
+
+    - name: Cache python virtualenv
+      uses: syphar/restore-virtualenv@v1.2
```
Comment (eduardostalinho) [inline .github/workflows/setupPythonVenv.yml:35]: I think we should have an input to enable/disable requirements cache.

**Review by eduardostalinho — COMMENTED** [review]
LGTM.

**Review by kotik-adjust — APPROVED** [review]

**Review by eduardostalinho — APPROVED** [review]

**Pull request merged**